sqlglot/schema.py

```diff
@@ -165,10 +165,25 @@ def column_names(self, table: exp.Table | str, only_visible: bool = False) -> t.
 
     def get_column_type(self, table: exp.Table | str, column: exp.Column) -> exp.DataType.Type:
         try:
-            schema_type = self.schema.get(table.name, {}).get(column.name).upper()  # type: ignore
+            if isinstance(table, exp.Table):
+                table_args = [
+                    table.text(p)
+                    for p in self.supported_table_args or self._get_table_args_from_table(table)
+                ]
+                table_schema = _nested_get(
+                    self.schema,
+                    *zip(self.supported_table_args, table_args),
+                    raise_on_missing=False,
+                )
+            else:
+                schema_type = self.schema.get(table, {})
```

Comment:
I don't think this line does anything, since we immediately set schema_type again? Unless I'm misreading something?

Reply:
Indeed, that was a bug for the str approach. Changed it now.